src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/probation/ProbationEventListenerIntTest.kt

```diff
@@ -142,6 +145,47 @@ class ProbationEventListenerIntTest : MessagingMultiNodeTestBase() {
     checkTelemetry(CPR_RECORD_CREATED, mapOf("SOURCE_SYSTEM" to "DELIUS", "CRN" to crn))
   }
 
+  @Test
+  fun `should log when message processing fails`() {
+    val crn = randomCRN()
+    stub500Response(crn, STARTED, "failure")
+    stub500Response(crn, STARTED, "failure")
+    stub500Response(crn, STARTED, "failure")
+    val crnType = PersonIdentifier("CRN", crn)
+    val personReference = PersonReference(listOf(crnType))
+
+    val domainEvent = DomainEvent(
+      eventType = NEW_OFFENDER_CREATED,
+      detailUrl = createDeliusDetailUrl(crn),
+      personReference = personReference,
+      additionalInformation = null,
+    )
+    val messageId = publishDomainEvent(NEW_OFFENDER_CREATED, domainEvent)
+
+    probationEventsQueue!!.sqsClient.purgeQueue(
+      PurgeQueueRequest.builder().queueUrl(probationEventsQueue!!.queueUrl).build(),
+    ).get()
+    probationEventsQueue!!.sqsDlqClient!!.purgeQueue(
+      PurgeQueueRequest.builder().queueUrl(probationEventsQueue!!.dlqUrl).build(),
+    ).get()
+    await.atMost(Duration.ofSeconds(2)) untilCallTo {
+      probationEventsQueue?.sqsClient?.countAllMessagesOnQueue(probationEventsQueue!!.queueUrl)?.get()
+    } matches { it == 0 }
+
+    await.atMost(Duration.ofSeconds(2)) untilCallTo {
+      probationEventsQueue?.sqsDlqClient?.countAllMessagesOnQueue(probationEventsQueue!!.dlqUrl!!)?.get()
+    } matches { it == 0 }
```

Comment:
A bit ungainly but we need to wait for message processing to complete and the message to be deleted off the queue